### PR TITLE
Code mirror for CC debugger

### PR DESCRIPTION
**Pull request by benrudolph**
@czue @wpride this makes the xml in the cc debugger a little prettier. :blowfish: 
screen shot 2015-11-22 at 5 58 55 pm

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -1,38 +1,39 @@
...
-        "nvd3": "1.1.10-beta"
-    },
-    "devDependencies": {
-        "chai": "3.3.0",
-        "mocha": "2.3.3",
-        "sinon": "http://sinonjs.org/releases/sinon-1.17.0.js",
-        "angular-mocks": "1.4.4"
-    },
-    "private": true
+  "name": "CommCareHQ",
```
Comment (benrudolph) [inline bower.json]: `bower install <lib> --save` automatically reformats your bower.json. this is also how most bower.json's are formatted so i decided to just leave it

**Comment by benrudolph** [comment]
for reference this is what it used to look like:
screen shot 2015-11-22 at 6 15 09 pm

**Comment by czue** [comment]
looks great!

**Comment by snopoke** [comment]
Seems like you're just using it for XML formatting. We're already using `syntaxhighlighter` in [other places](https://github.com/dimagi/commcare-hq/blob/ad727a881c02f7dadbb120a2c161a0b06a1ce1f4/corehq/apps/app_manager/templates/app_manager/form_view_base.html#L13-L13). Is `codemirror` that much better\?

**Comment by benrudolph** [comment]
@snopoke interesting, didn't know we used that. i thought codemirror was defactor based on UCR and the raw_couch view but seeing this makes me think otherwise. i'll update. also probably a good candidate to add to bower

**Comment by snopoke** [comment]
👍 was just suggesting we consider what we want to use for this kind of
thing on try to stick to one. I see lots of other places using `codemirror`
but nowhere else using syntaxhighlighter so perhaps it should be the other
way around and we drop the use of syntaxhighlighter.

On 23 November 2015 at 15:54, Ben <notifications@github.com> wrote:

> @snopoke <https://github.com/snopoke> interesting, didn't know we used
> that. i thought codemirror was defactor based on UCR and the raw_couch view
> but seeing this makes me think otherwise. i'll update. also probably a good
> candidate to add to bower
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/commcare-hq/pull/9225#issuecomment-158938082>.
>



-- 
Simon Kelly
Senior Engineer | Dimagi South Africa

**Comment by wpride** [comment]
@benrudolph Nice! this is so sexy

**Pull request merged**